File: client/src/types/voucher.ts
export type VoucherDiscountType = 'Percentage' | 'FixedAmount'

export interface Voucher {
  id: string
  code: string
  discountType: VoucherDiscountType
  discountValue: number
  minOrderAmount: number
  maxUsageCount: number | null
  usedCount: number
  maxUsagePerUser: number
  expiresAt: string
  isActive: boolean
}

export interface CreateVoucherRequest {
  code: string
  discountType: VoucherDiscountType
  discountValue: number
  minOrderAmount: number
  maxUsageCount?: number | null
  maxUsagePerUser: number
  expiresAt: string
  isActive: boolean
}

export type UpdateVoucherRequest = CreateVoucherRequest

export interface VoucherValidationResult {
  voucherId: string
  code: string
  subtotal: number
  discountAmount: number
  finalTotal: number
}

export interface PagedResult<T> {
  items: T[]
  totalCount: number
  page: number
  pageSize: number
  totalPages: number
}

File: src/Domain/Entities/Voucher.cs
using HungStore.Domain.Common;
using HungStore.Domain.Enums;

namespace HungStore.Domain.Entities;

public class Voucher : BaseEntity
{
    public string Code { get; set; } = string.Empty;
    public VoucherDiscountType DiscountType { get; set; }
    public decimal DiscountValue { get; set; }
    public decimal MinOrderAmount { get; set; }
    public int? MaxUsageCount { get; set; }
    public int UsedCount { get; set; }
    public int MaxUsagePerUser { get; set; } = 1;
    public DateTime ExpiresAt { get; set; }
    public bool IsActive { get; set; } = true;
}
